providers/base/api/handlers/api_query_handler.go

```diff
@@ -65,6 +65,21 @@ func NewAPIQueryHandler[K providertypes.ResponseKey, V providertypes.ResponseVal
 	requestHandler RequestHandler,
 	apiHandler APIDataHandler[K, V],
 	metrics metrics.APIMetrics,
+) (APIQueryHandler[K, V], error) {
+	priceFetcher, err := NewRestAPIFetcher(requestHandler, apiHandler, metrics, cfg, logger)
+	if err != nil {
+		return nil, fmt.Errorf("failed to create price fetcher: %w", err)
+	}
+
+	return NewAPIQueryHandlerWithPriceFetcher(logger, cfg, priceFetcher, metrics)
+}
+
+// NewAPIQueryHandlerWithPriceFetcher creates a new APIQueryHandler with a custom price fetcher.
+func NewAPIQueryHandlerWithPriceFetcher[K providertypes.ResponseKey, V providertypes.ResponseValue](
```

Comment:
I would prefer we remove this new function and add variadic `Options` to `NewAPIQueryHandler`.  Then a `WithPriceFetcher` option can be passed in

Reply:
I don't think the optional `PriceFetcher` works in all cases? As there are two ways to approach this:
1. We pass in an un-needed `apiDataHandler`, in this case, if I'm making a defi provider (raydium / uniswap), we j pass in a nil `apiDataHandler`
2. We don't have the `apiDataHandler` as an argument, in this case, the `apiPriceFetcher` is not really optional?
Ultimately, the `QueryHandler` explicitly depends on the `PriceFetcher` so having it be an optional parameter does not seem like the correct move here?

I may be mis-understanding tho, why is the `Option` argument preferred?

Reply:
I agree with alex that switching to options is eventually going to be better as we implement more implementations of the APIQueryHandler, but i think this is fine for the time being.